task1.py

```diff
@@ -0,0 +1,55 @@
+import sys
+
+
+def convert_dict_to_string(dict):
+    str_output = ""
+    for keys, values in dict.items():
+        str_output += values
+
+    return str_output
+
+
+def remove_all_except_first(str_input):
+    dic_removed_duplicates = {}
+    for c in str_input:
+        if str(c).upper() not in dic_removed_duplicates:
+            dic_removed_duplicates[str(c).upper()] = c
+
+    return convert_dict_to_string(dic_removed_duplicates)
+
+
+def remove_first_occurrence(str_input):
+    slow_index = 0
+    fast_index = 1
+    count = len(str_input)
+    lstDeletedCharacters = list()
+
+    while slow_index < count:
+        if slow_index != fast_index and str_input[slow_index].upper() == str_input[fast_index].upper() \
+                and lstDeletedCharacters.count(str_input[slow_index]) == 0:
+            lstDeletedCharacters.append(str_input[slow_index])
+            str_input = str_input[0:slow_index] + '*' + str_input[slow_index + 1:]
+
+        fast_index += 1
+
+        if fast_index >= count:
+            slow_index += 1
+            fast_index = 0
+
+    str_input = str_input.replace("*", "")
+
+    return str_input
```

Comment:
That solution works but is not very readable. Use meaningful names and comments to describe the code. See for example the following version:

```python 
def remove_first_occurrence(word):
    # create a character count first 
    character_count = {}
    for character in word:
        if character not in character_count:
            character_count[character] = 1
        else:
            character_count[character] += 1

    for character, count in character_count.items():
        # if a character is repeated more than once, remove its first occurence
        if count > 1:
            word = word.replace(character, '', 1)
    
    return word
```